### PR TITLE
Add conversions between Moonlight and Phoenix to `wallet-core` and test them

### PR DESCRIPTION
**Pull request by ureeves**
As advertised, this PR introduces two functions in `wallet-core` to be able to convert Dusk between Moonlight and Phoenix. The functions are:

- `moonlight_to_phoenix`
- `phoenix_to_moonlight`

The tests occur in `rusk` in the context of block ingestion, so we're sure that the transactions these two functions create can be ingested properly by the chain.

As a consequence, a totally generic `moonlight` function is created, mirroring the `phoenix` function, that allows for creating a Moonlight transaction in a similar way to Phoenix.

### CODE REVIEW

File: rusk/tests/services/conversion.rs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        .get_balance(0)
+        .expect("Getting phoenix balance should succeed");
+
+    assert_eq!(
+        moonlight_account.balance, INITIAL_MOONLIGHT_BALANCE - CONVERT_VALUE - gas_spent,
+        "The Moonlight account must have its initial value minus the converted amount and gas spent"
+    );
+    assert_eq!(
+        phoenix_balance.value, INITIAL_PHOENIX_BALANCE + CONVERT_VALUE,
+        "The Phoenix notes should must be of their value plus the converted amount"
```
Comment (HDauven) [inline rusk/tests/services/conversion.rs:155]: "The Phoenix notes should have its initial balance plus the converted amount" or something similar

File: wallet-core/src/transaction.rs [hunk 2]
```diff
@@ -46,6 +47,7 @@ use execution_core::{
 /// - the `inputs` vector is either empty or larger than 4 elements
 /// - the `inputs` vector contains duplicate `Note`s
 /// - the `Prove` trait is implemented incorrectly
+/// - the Memo provided with `data` is too large
```
Comment (HDauven) [inline wallet-core/src/transaction.rs:50]: Nice catch

**Review by HDauven — DISMISSED** [review]
LGTM

File: rusk/tests/services/conversion.rs (hunk 1 above)
Comment (ureeves) [reply rusk/tests/services/conversion.rs]: Fixed

**Review by HDauven — APPROVED** [review]

**Pull request merged**